### ISSUE TITLE
Tons of duplicate entries on database

### ISSUE THREAD

**Issue by fernando0010** [clementine-player/Clementine#6777]
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [x] I checked the issue tracker for similar issues
- [x] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [x] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: Ubuntu 20.04.1
- Clementine version: 1.4.0rc1-296-g68d375c43~focal

### Expected behaviour / actual behaviour
All of a sudden, Clementine has duplicated the entries for lots of files on my database, each one referring to the same mp3 but with separate play count, rating, etc. Is there any chance to eliminate the duplicates without wiping the DB and/or losing track statistics?

### Steps to reproduce the problem (only for bugs)
Attaching screen capture to illustrate the problem:
![Captura de pantalla de 2020-08-21 13-16-57](https://user-images.githubusercontent.com/20421920/90912455-058c0380-e3b1-11ea-9aa0-552a55bce6dc.png)

**Comment by jbroadus** [comment]
On thing that I notice is that `FindSongsInSubdirectory` will return an empty song list if an error occurs, so the caller wouldn't be able to differentiate between an error and an empty directory. I'm not sure if temporary sqlite errors could occur.

Besides that, is there anything interesting about the layout of your library? symlinks, etc?

**Comment by fernando0010** [comment]
Nope, a couple of (large) directories and that's it. I solved the duplicate entries bug with the SQL commands from the link above, but now I find that new folders aren't getting indexed into the DB :(
